File: 2124.检查是否所有-a-都在-b-之前.js
/*
 * @lc app=leetcode.cn id=2124 lang=javascript
 *
 * [2124] 检查是否所有 A 都在 B 之前
 */

// @lc code=start
/**
 * @param {string} s
 * @return {boolean}
 */
var checkString = function(s) {
    let bSeen = false;
    for (let char of s) {
        if (char === 'b') {
            bSeen = true;
        } else if (char === 'a' && bSeen) {
            return false;
        }
    }
    return true;
};
// @lc code=end

File: 2124.检查是否所有-a-都在-b-之前.java
/*
 * @lc app=leetcode.cn id=2124 lang=java
 *
 * [2124] 检查是否所有 A 都在 B 之前
 */

// @lc code=start
class Solution {
    public boolean checkString(String s) {
        boolean bSeen = false;
        for (char c : s.toCharArray()) {
            if (c == 'b') {
                bSeen = true;
            } else if (c == 'a' && bSeen) {
                return false;
            }
        }
        return true;
    }
}
// @lc code=end

